qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,363 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Second-order Møller-Plesset perturbation theory (MP2) computation."""
+
+from __future__ import annotations
+
+import numpy as np
+from qiskit_nature.exceptions import QiskitNatureError
+
+from qiskit_nature.properties.second_quantization.electronic import ElectronicEnergy
+from qiskit_nature.properties.second_quantization.electronic.bases import ElectronicBasis
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.properties.second_quantization.electronic.integrals.electronic_integrals import (
+    ElectronicIntegrals,
+)
+from qiskit_nature.properties.second_quantization.second_quantized_property import (
+    GroupedSecondQuantizedProperty,
+)
+
+from .initial_point import InitialPoint
+
+
+class MP2InitialPoint(InitialPoint):
+    """Second-order Møller-Plesset perturbation theory (MP2) computation.
+
+    A class for computing the second-order Møller-Plesset perturbation theory (MP2) corrections.
+    This is intended for use as an initial point for the VQE parameters in combination with a UCC
+    ansatz.
+
+    :class:`MP2InitialPoint` requires two-body molecular orbital electronic integrals and orbital
+    energies from
+    :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`,
+    which should be passed in via the :attr:`grouped_property` attribute.
+
+    :class:`MP2InitialPoint` also requires the :attr:`excitation list` from the :attr:`~ansatz`
+    to ensure that the coefficients are mapped correctly in the initial point array. This can be
+    overwritten by setting :attr:`~excitation_list` directly.
+
+    The coefficients and energy corrections are computed using :meth:`~compute`, which requires
+    :attr:`grouped_property` and :attr:`~ansatz` to be set or passed as arguments to
+    :meth:`~compute`.
+
+    Following computation, the initial point array can be extracted via `:meth:to_numpy_array`.
+    Array elements with indices corresponding double excitations in the :attr:`~excitation_list`
+    will use the computed MP2 coefficient, while those that correspond to single, triple, or
+    higher excitations will be zero.
+
+    The array of energy corrections for each excitation can be
+    recovered using :meth:`get_energy_corrections`. The overall correction is got via
```

Comment:
```suggestion
    recovered using :meth:`get_energy_corrections`. The overall correction can be obtained via
```
You may need to reformat slightly to avoid the line length limit being exceeded